sneck/Deck.py

```diff
@@ -295,47 +386,78 @@ class Type(Enum):
     }
 
     # References to the users and labels dictionaries.
-
     def __init__(self, card: dict, bid: int, labels: dict[DeckLabel], users: dict[DeckUser], api: DeckAPI, updateAttachments: bool = True) -> None:
-        # IDs of the parents of the card
-        self.__board_id = bid
-        self.__stack_id = card['stackId']
-
-        # References to the label and user dictionaries and the API.
-        self.__users_dict = users
-        self.__labels_dict = labels
-        self.__api = api
-
-        # Attributes of the card
-        self.__id = card['id']
-        self.__tag = card['ETag']
-        self.__type = self.__types[card['type']]
-        self.__order = card['order']
-        self.__title = card['title']
-        self.__description = card['description'].strip()
-        self.__labels = [labels[label['ETag']] for label in card['labels']]
-        self.__archived = card['archived']
-        self.__unread_comments_count = card['commentsUnread']
-
-        # Timestamps
-        self.__creation_time = dt.fromtimestamp(card['createdAt']).astimezone(tz.utc)
-        self.__last_edited_time = dt.fromtimestamp(card['lastModified']).astimezone(tz.utc)
-        self.__deletion_time = (dt.fromtimestamp(card['deletedAt']).astimezone(tz.utc)
-                                if card['deletedAt'] != 0 else None)
-        self.__card_due_time = (None if card['duedate'] is None else
-                                dt.strptime(card['duedate'], '%Y-%m-%dT%H:%M:%S%z').astimezone(tz.utc))
-
-        # Users
-        self.__assigned_users = [users[assignee['participant']['uid']] for assignee in card['assignedUsers']]
-        self.__owner = users[card['owner']['uid']]
-        self.__last_editor = users[card['lastEditor']] if card['lastEditor'] is not None else None
-
-        # Card attachments
-        self.__attachments = ({attachment['id']: DeckAttachment(attachment, self.__stack_id, self.__board_id, users) for attachment in
-                              api.request(f'boards/{bid}/stacks/{self.__stack_id}/cards/{self.__id}/attachments')}
-                              if card['attachmentCount'] > 0 else {})
-        
-        print(f'[UPDATE] Created card {self.__id}.')
+        if card is not None and bid is not None and labels is not None and users is not None:
```

Comment:
I would change the condition to `if card is not None` since the bid, user dictionary and labels might be known a priori (for example if the workflow is something like `deck.get_board(board_id).add_card()` from the client perspective, with the board). This is not really required as the `DeckBoard` might create the object with null elements and assign them afterwards, but it does skip a step down the line

Reply:
I don't fully understand the reason behind this change, since if the bid and the user and label dictionaries are known, we are sure that they are not "None". And when  `deck.get_board(board_id).add_card()` is called, it accepts the sid and a `DeckCard` object (`deck.get_board(board_id).add_card(stack_id, deck_card)`). 
Though of course I might have missed your point completely :)